src/Microsoft.Framework.Configuration.File/FileConfigurationbuilderExtensions.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Framework.Configuration
+{
+    public static class FileConfigurationBuilderExtensions
+    {
+        /// <summary>
+        /// Sets the base path in <see cref="IConfigurationBuilder"/> for file based providers.
+        /// </summary>
+        /// <param name="builder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="basePath">Absolute path of file based providers.
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder SetBasePath(this IConfigurationBuilder builder, string basePath)
```

Comment:
Should there be a `GetBasePath()` companion to this? Trivial to implement, and appears more symmetrical (rare to get a set without a get). I realize our own code won't need this, but it might be nice for other consumers that might just want the path to do their own manipulation (rather than use the resolve helper method).